### PR TITLE
[EOC-276] archived sack view disabled restore and delete for viewer and member

### PR DESCRIPTION
**Pull request by alemikolo**
I merged branch EOC-278 and EOC-281 to this branch to avoid some errors which were fixed there.
1. The viewer can't get archived list metadata and can't access archived list view - they get not-found page.

2. The member can get and see archived lists and can access archived list view but they get only info that this list is archived. They can't perform restore and archive action:
Screenshot 2019-07-08_12-37-35-152

3. The owner can get and see archived lists and can access archived list view. They get our previous archived view and can perform restore and archive action:
Screenshot 2019-07-08_12-37-48-693

### CODE REVIEW

File: client/src/modules/cohort/index.jsx [hunk 1]
```diff
@@ -81,17 +90,32 @@ class Cohort extends PureComponent {
...
-      })
-      .finally(() => {
         this.handleBreadcrumbs();
         this.setState({ pendingForDetails: false });
+      })
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pendingForDetails: false });
+        }
+        noOp();
```
Comment (adamklepacz) [inline client/src/modules/cohort/index.jsx]: @marekrozmus   
is no op() needed?

File: client/src/modules/list/components/ArchivedList.jsx [hunk 2]
```diff
@@ -33,6 +34,7 @@ class ArchivedList extends PureComponent {
 }
 
 ArchivedList.propTypes = {
+  isOwner: PropTypes.bool,
```
Comment (adamklepacz) [inline client/src/modules/list/components/ArchivedList.jsx]: shouldn't be required?

File: client/src/modules/list/index.jsx [hunk 3]
```diff
@@ -60,12 +61,18 @@ class List extends Component {
...
-    this.fetchData().finally(() => {
-      this.setState({ pendingForDetails: false });
-      this.handleBreadcrumbs();
-      this.handleSocketConnection();
-      this.receiveWSEvents();
-    });
+    this.fetchData()
+      .then(() => {
+        this.handleBreadcrumbs();
+        this.handleSocketConnection();
```
Comment (adamklepacz) [inline client/src/modules/list/index.jsx]: remove socket connection from here as we have HOC component right now. 
and change hame to handleRoomConnection() cause you'll be only joining a room right now

File: client/src/common/components/ArchivedMessage.jsx [hunk 4]
```diff
@@ -101,6 +104,7 @@ class ArchivedMessage extends PureComponent {
 
 ArchivedMessage.propTypes = {
   intl: IntlPropType.isRequired,
+  isOwner: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/common/components/ArchivedMessage.jsx]: Boolean values don't have to be required if you don't pass them they will be falsy

File: client/src/modules/cohort/components/ArchivedCohort.jsx [hunk 5]
```diff
@@ -37,6 +38,7 @@ class ArchivedCohort extends PureComponent {
 
 ArchivedCohort.propTypes = {
   cohortId: PropTypes.string.isRequired,
+  isOwner: PropTypes.bool,
```
[the remark this answers is not in the document]
Comment (alemikolo) [reply client/src/modules/cohort/components/ArchivedCohort.jsx]: Boolean values don't have to be required if you don't pass them they will be falsy

File: client/src/modules/list/components/ArchivedList.jsx (hunk 2 above)
Comment (alemikolo) [reply client/src/modules/list/components/ArchivedList.jsx]: Boolean values don't have to be required if you don't pass them they will be falsy

File: client/src/modules/list/index.jsx (hunk 3 above)
Comment (alemikolo) [reply client/src/modules/list/index.jsx]: this old branch, there is no this HOC here, this place should be changed in this task EOC-283,

**Pull request merged**